lib/bolt/transport/base.rb

```diff
@@ -87,6 +87,15 @@ def select_implementation(target, task)
         impl
       end
 
+      def select_interpreter(executable, interpreters)
```

Comment:
I thought about adjusting for case here as well? For example  `.rb,rb,.RB,.rb` would all map to ruby interpreter specified? Is that too much "magic"? Is there a sane workflow that someone would choose to map `.rb` to one interpreter and `.RB` to another?

Reply:
The ticket states "extension name to absolute path of an executable (exact match to extension)" which I interpret to mean do not try to ignore case.

Reply:
I think honoring case is the right thing to do here. But we should normalize the extensions (to either always include 
or never include a `.`)  when processing the config, so we only have to look up a single key. Because `File.extname()` return an extension containing a `.`, it probably makes sense to normalize the config option to always include a `.`